hbase-server/src/main/java/org/apache/hadoop/hbase/wal/RecoveredEditsOutputSink.java

```diff
@@ -61,13 +61,13 @@
  * Class that manages the output streams from the log splitting process.
  */
 @InterfaceAudience.Private
-public class LogRecoveredEditsOutputSink extends OutputSink {
-  private static final Logger LOG = LoggerFactory.getLogger(LogRecoveredEditsOutputSink.class);
-  private WALSplitter walSplitter;
-  private FileSystem walFS;
-  private Configuration conf;
+public class RecoveredEditsOutputSink extends OutputSink {
+  private static final Logger LOG = LoggerFactory.getLogger(RecoveredEditsOutputSink.class);
+  protected WALSplitter walSplitter;
+  protected FileSystem walFS;
```

Comment:
And I believe this will introduce problems. Think of HBase on S3, where we will still write WAL to HDFS, but HFile will be on S3. So for writing HFiles we should use the HFile FileSystem instead of WAL FileSystem.

Reply:
Yes. The new HFile should write to the HFile file system.